tests/model_monitoring/test_applications/test_base.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from mlrun.model_monitoring.applications import (
+    ModelMonitoringApplicationBase,
+    ModelMonitoringApplicationBaseV2,
+)
+
+
+def test_deprecation_v1_instantiation() -> None:
+    class DeprecatedApp(ModelMonitoringApplicationBase):
+        def do_tracking(self):
+            pass
+
+    with pytest.warns(DeprecationWarning):
+        DeprecatedApp()
+
+
+@pytest.mark.filterwarnings("error")
+def test_no_deprecation_v2_instantiation() -> None:
+    class AppV2(ModelMonitoringApplicationBaseV2):
+        def do_tracking(self):
```

Comment:
```suggestion
        def do_tracking(self, **kwargs):
```

I suggest to update the signature of the abstraction method to avoid annoying warnings from the IDE:
`Signature of method 'AppV2.do_tracking()' does not match signature of the base method in class 'ModelMonitoringApplicationBaseV2'`